### PR TITLE
Split pytato.array

### PR DESCRIPTION
**Pull request by kaushikcfd**
Moves
- math related interfaces into `pytato.cmath`
- reduction related interface into `pytato.reductions`.

For best reviewing experience please look at the patch on a commit-by-commit basis.

### CODE REVIEW

File: pytato/cmath.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
+from __future__ import annotations
```
Comment (inducer) [inline pytato/cmath.py:1]: Why `cmath` (as opposed to, say, `ufunc` or `elwise`)?

Comment (kaushikcfd) [reply pytato/cmath.py:1]: I thought a lot of routines would fall under `pt.unfuc` and was shooting for a finer-grained separation.

Comment (inducer) [reply pytato/cmath.py:1]: Sure. It's internal anyway, so why not. It's better than nothing. :)

**Comment by inducer** [comment]
x-ref: #52

**Comment by inducer** [comment]
LGTM. Thanks for working on this!

**Pull request merged**